lc2fen/infer_pieces.py

```diff
@@ -88,49 +169,70 @@ def __sort_pieces_list(_pieces_probs_sort):
     ]
 
 
-def __max_piece(tops):
-    """
-    The function takes a nested list of sorted piece probabilities(tops) and
-    Returns the index of the piece with max probability.
+def _piece_with_highest_prob(top_probs_by_type: list[tuple[list, int]]) -> int:
+    """Determine the piece that has the highest piece probability across the entire chessboard.
```

Comment:
Thanks! For conciseness, I chose not to mention the subtlety that those already-determined squares were not taken into account when determining the highest piece probability "across the entire chessboard." (As with most technical/mathematical issues, the more you want something to be clearly explained, the more words you'll typically have to use and the more risk you run of making the resulting statement too complex and only understandable to yourself...)

I think the paper figure would help (it might not make immediate sense to the reader but I think it might help the reader understand the code a bit more easily)! Just two small issues:
(1) I've renamed `tops` to `top_probs_by_type` but `tops` is used in the figure
(2) I'd suggest pointing the "Probability" and "Square index" labels to perhaps the `[0.95, 51]` element and pointing that element to the square on the board:
![image](https://github.com/davidmallasen/LiveChess2FEN/assets/51491203/9578fa63-4ce8-49ba-ba5b-7af40e39104b)

Reply:
Done! It is in the `docs` folder. I also added the source of the diagram in case we want to make any modifications to it in the future. I created it using [draw.io](draw.io).

Reply:
I've added the figure reference